light-client/src/peer_list.rs

```diff
@@ -1,156 +1,206 @@
 use crate::{
     bail,
     errors::{Error, ErrorKind},
-    supervisor::Instance,
     types::PeerId,
 };
 
 use contracts::{post, pre};
-use std::collections::{HashMap, HashSet};
+use std::collections::{BTreeSet, HashMap};
 
-/// A mapping from PeerIds to Light Client instances.
-/// Keeps track of which peer is deemed the primary peer.
+/// A generic container mapping `PeerId`s to some type `T`,
+/// which keeps track of the primary peer, witnesses, full nodes,
+/// and faulty nodes. Provides lifecycle methods to swap the primary,
+/// mark witnesses as faulty, and maintains an `invariant` for
+/// correctness.
 #[derive(Debug)]
-pub struct PeerList {
-    instances: HashMap<PeerId, Instance>,
+pub struct PeerList<T> {
+    values: HashMap<PeerId, T>,
```

Comment:
Why the quite generic rename here?

Reply:
Since the peer list can now hold to any datatype rather than just light client instances, I thought `values` was more appropriate than `instances`, but I am open to suggestions.